### PR TITLE
Implement tabs below content

### PR DESCRIPTION
**Pull request by onemen**
@117649 can you test it?

### CODE REVIEW

**Comment by 117649** [comment]
> 
> 
> @117649 can you test it?

I think so. Just need to find some time first.
I think I should also try it on waterfox since that browser have an implement of it own.

**Comment by 117649** [comment]
图片
Dark theme support are needed.

**Comment by 117649** [comment]
图片
@onemen Enter customize mode resulting this.
Everything is covered its kinda series.

**Comment by 117649** [comment]
图片
Missing window buttons.

**Comment by onemen** [comment]
what theme and Firefox version you are using ?

**Comment by 117649** [comment]
> 
> 
> what theme and Firefox version you are using ?
> 
> and what is your system colors in windows 10

91.0b9 dark theme enable.
windows is on light.

**Comment by onemen** [comment]
> 
> 
> > what theme and Firefox version you are using ?
> > and what is your system colors in windows 10
> 
> 91.0b9 dark theme enabled.
> windows is on light.

I can't reproduce this issue, maybe it related to difference in our settings
Please post a file with your Tabmix preferences.
Do you see this issue when there is only one tab?, or when tab is overflowing?

**Comment by 117649** [comment]
> 
> 
> > > what theme and Firefox version you are using ?
> > > and what is your system colors in windows 10
> > 
> > 
> > 91.0b9 dark theme enabled.
> > windows is on light.
> 
> I can't reproduce this issue, maybe it related to difference in our settings
> Please post a file with your Tabmix preferences.
> Do you see this issue when there is only one tab?, or when tab is overflowing?

Which issue?

**Comment by onemen** [comment]
> Which issue?

Missing window buttons.

I see it now. Your Firefox settings is to show  window buttons on TabsToolbar, (you hide the "Menu Bar")

I need to add some css rules to always show window buttons on top when the tabs are below content.

**Comment by 117649** [comment]
Tab bar is forced back to above in customize mode. Waterfox G3 don't have this problem.
And for WF TMP's option for tab bar position have no effect while **about:preferences** isn't open.

**Comment by onemen** [comment]
> Tab bar is forced back to above in customize mode. Waterfox G3 don't have this problem.

I know, i had some issues with customize mode in the past in. I will have to check again if the tabs can be in the bottom in customize mode.

> And for WF TMP's option for tab bar position have no effect while **about:preferences** isn't open.

I don't understand what you mean by  **TMP's option**, if you refer to ```extensions.tabmix.tabBarPosition```, In WF it linked to WF preference ```browser.tabBar.position``` also Tabmix options window linked to ```browser.tabBar.position```, see https://github.com/onemen/TabMixPlus/pull/60/commits/f6a1a47f2e41ed61b431440e433704b45183e44b

**Comment by 117649** [comment]
> I don't understand what you mean by **TMP's option**, if you refer to `extensions.tabmix.tabBarPosition`, In WF it linked to WF preference `browser.tabBar.position` also Tabmix options window linked to `browser.tabBar.position`, see [f6a1a47](https://github.com/onemen/TabMixPlus/commit/f6a1a47f2e41ed61b431440e433704b45183e44b)

The option dialog window.
On my system this option have no effect if there is not opened **about:preference** instance.
图片

File: addon/chrome/content/preferences/appearance.js [hunk 1]
```diff
@@ -59,6 +59,32 @@ var gAppearancePane = {
...
+      position.setAttribute("preference", "browser.tabBar.position");
+      position.setAttribute("value", Services.prefs.getCharPref("browser.tabBar.position"));
+      const preferences = document.documentElement.querySelector("preferences");
+      preferences.appendChild(MozXULElement.parseXULToFragment(
+        `<preference id="browser.tabBar.position"
+                     name="browser.tabBar.position"
+                     type="string"/>`
+      ));
+    }
+
```
Comment (onemen) [inline addon/chrome/content/preferences/appearance.js:87]: > The option dialog window.
> On my system this option have no effect if there is not an opened **about:preference** instance.

@117649 ,
this code work for me, maybe there is some race condition?
I don't know why ```about:preferences``` change anything
check if you can find if calling this code early help

or maybe try to change this
```diff
      position.setAttribute("preference", "browser.tabBar.position");
      position.setAttribute("value", Services.prefs.getCharPref("browser.tabBar.position"));
-     const preferences = document.documentElement.querySelector("preferences");
+     const preferences = $("paneAppearance").querySelector("preferences");
      preferences.appendChild(MozXULElement.parseXULToFragment(
```

**Comment by onemen** [comment]
got it

```
        `<preference id="browser.tabBar.position"
                     name="browser.tabBar.position"
                     type="string"
                     onchange="window.opener.moveTabBar();"/>`
```

File: addon/chrome/content/preferences/appearance.js (hunk 1 above)
Comment (onemen) [reply addon/chrome/content/preferences/appearance.js:87]: got it !

```
        `<preference id="browser.tabBar.position"
                     name="browser.tabBar.position"
                     type="string"
                     onchange="window.opener.moveTabBar();"/>`
```

**Pull request merged**